### PR TITLE
PR: Use text instead of key in the editor close bracket extension

### LINKED ISSUE #9749: No curly brackets on Spyder 4.0.0b3 on macOS with a French keyboard
### Issue Report Checklist

* [x ] Searched the [issues page](https://github.com/spyder-ide/spyder/issues?q=is%3Aissue) for similar reports
* [x ] Read the relevant sections of the [Spyder Troubleshooting Guide](https://github.com/spyder-ide/spyder/wiki/Troubleshooting-Guide-and-FAQ) and followed its advice
* [x ] Reproduced the issue after updating with ``conda update spyder`` (or ``pip``, if not using Anaconda)
* [x ] Could not reproduce inside ``jupyter qtconsole`` (if console-related)
* [x ] Tried basic troubleshooting (if a bug/error)
    * [x ] Restarted Spyder
    * [x ] Reset preferences with ``spyder --reset``
    * [ ] Reinstalled the latest version of [Anaconda](https://www.anaconda.com/download/)
    * [ ] Tried the other applicable steps from the Troubleshooting Guide
* [x ] Completed the **Problem Description**, **Steps to Reproduce** and **Version** sections below

## Problem Description
No curly brackets on Mac

### What steps reproduce the problem?

Type { }

### What is the expected output? What do you see instead?
I see only parentheses ( )

## Versions

* Spyder version: 4.0.0beta3 (same in beta2)
* Python version: 3.7.3
* Qt version: 5.9.7
* PyQt version: 5.9.2
* Operating System name/version: MacOS 10.13.6

### Dependencies

```
pygments >=2.0    :  2.4.2 (OK)
qdarkstyle >=2.6.4:  2.7 (OK)
sphinx >=0.6.6    :  2.1.2 (OK)
psutil >=0.3      :  5.6.3 (OK)
pyls >=0.27.0     :  0.27.0 (OK)
nbconvert >=4.0   :  5.5.0 (OK)
pandas >=0.13.1   :  None (NOK)
numpy >=1.7       :  None (NOK)
sympy >=0.7.3     :  None (NOK)
cython >=0.21     :  None (NOK)
qtconsole >=4.5.0 :  4.5.1 (OK)
IPython >=4.0     :  6.5.0 (OK)
matplotlib >=2.0.0:  None (NOK)
pylint >=0.25     :  2.3.1 (OK)
```

### PR DESCRIPTION
**Pull request by jnsebgosselin**
## Description of Changes

* [x] Included a screenshot or animation (if affecting the UI, see [Licecap](https://www.cockos.com/licecap/))

curly

Automatic insertion of parentheses, brackets and braces should be done against the text of the key press event instead of the key in the `CloseBracketsExtension`.

This is because on some international keyboards, the key that was pressed does not always correspond to the text when some key modifiers are used.

For example, on the macOS with a french keyboard, the key combination `Alt + (` results in the key press event text `{`, while the key would be `Qt.Key_ParenLeft`.

PS: I have no mean to test that this is actually fixing #9749, so please review carefully :)

### Issue(s) Resolved

Fixes #9749

### CODE REVIEW

**Comment by steff456** [comment]
I just tested with the french keyboard and it is working as expected. As well, I added a gif with the corrected feature and re-run the failed tests (failing by LSP tests)

File: spyder/plugins/editor/extensions/closebrackets.py [hunk 1]
```diff
@@ -35,10 +29,10 @@ def _on_key_pressed(self, event):
         if event.isAccepted():
             return
 
-        key = event.key()
-        if key in self.BRACKETS_CHAR and self.enabled:
+        char = event.text()
```
Comment (goanpeca) [inline spyder/plugins/editor/extensions/closebrackets.py]: I would just add a comment here to highlight why we use `event.text()`

Something like. `# See spyder-ide/spyder#9749` to keep the code informed for future devs

**Comment by jnsebgosselin** [comment]
Thank you both @steff456 and @goanpeca with your help and review with this one.

**Comment by jproy** [comment]
Thanks ! I will install Spyder 4beta again.

**Comment by ccordoba12** [comment]
@jproy, this will be part of beta4, to be released at the end of this month.

**Comment by jproy** [comment]
Ok @ccordoba12. I will be able to come back around Aug 15 to taste the 4beta. Thanks to @jnsebgosselin and @steff456 !

**Pull request merged**